### PR TITLE
Homework4: Mikalai Hutsalenka

### PR DESCRIPTION
**Pull request by vorongn**

### CODE REVIEW

File: src/hw04_task02_cities.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Novgorod
+
+Выходные данные
+Ukraine
+Russia
+Russia
+"""
+
+
+strani = {}
```
Comment (belovmd) [inline src/hw04_task02_cities.py]: countries

File: src/hw04_task02_cities.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+count_strani = int(input('Введите количество стран: '))
+for numm in range(count_strani):
```
Comment (belovmd) [inline src/hw04_task02_cities.py]: where num is used? 
for _ in ...

File: src/hw04_task02_cities.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    strana, *goroda = input('страна город город ......:').split()
+    strani[strana] = goroda
```
Comment (belovmd) [inline src/hw04_task02_cities.py]: cities

File: src/hw04_task02_cities.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+"""
```
Comment (belovmd) [inline src/hw04_task02_cities.py:1]: what about Brest in Belarus and France?

File: src/hw04_task06_unique_words.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+"""
+Во входной строке записан текст.
+Словом считается последовательность непробельных символов идущих подряд,
+слова разделены одним или большим числом пробелов или символами конца
+строки. Определите, сколько различных слов содержится в этом тексте.
+"""
+
+
+import re
+stroka01 = "Lorem Ipsum is simply dummy text of the" \
```
Comment (belovmd) [inline src/hw04_task06_unique_words.py]: we can use 
```
("string"
 "string")
```
except \ symbol

File: src/hw04_task06_unique_words.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+           " printing and typesetting industry. " \
+           "Lorem Ipsum has been the industry's " \
+           "standard dummy text ever since the 1500s, " \
+           "when an unknown printer took a galley of " \
+           "type and scrambled it to make a type specimen book."
+
+stroka02 = set(re.sub(r"[^A-Za-zА-Яа-я]+", ' ', stroka01).split())
```
Comment (belovmd) [inline src/hw04_task06_unique_words.py]: wouldn't strip be enough?

File: src/hw04_task07_evklid.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+Повторяем пока остаток не станет =0.
+"""
+number01 = int(input('число1:'))
+number02 = int(input('число2:'))
+while number01 != 0 and number02 != 0:
+    if number01 > number02:
+        number01 = number01 % number02
+    else:
+        number02 = number02 % number01
+print(number01 + number02)
```
Comment (belovmd) [inline src/hw04_task07_evklid.py]: it's a cheat. Algorithm doesn't expect result in a sum
you can use ternary operator instead

**Review by belovmd — CHANGES_REQUESTED** [review]
Please take a look at these issues

**Comment by vorongn** [comment]
fixed

File: src/hw04_task02_cities.py (hunk 4 above)
Comment (vorongn) [reply src/hw04_task02_cities.py:1]: Conclusion changed to city-country. Now it’s clear where the cities are found. Duplicate names are displayed several times, but the country is indicated.

File: src/hw04_task07_evklid.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+Вычислите их наибольший общий делитель
+при помощи алгоритма Евклида
+(мы не знаем функции и рекурсию).
+----------------------------------
+"""
+
+number01 = int(input('число1:'))
+number02 = int(input('число2:'))
+while number02 != 0:
+    number01, number02 = number02, number01 % number02
```
Comment (vorongn) [inline src/hw04_task07_evklid.py:12]: fixed

File: src/hw04_task06_unique_words.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import re
+
+stroka01 = ("Lorem Ipsum is simply dummy text of the "
+            "printing and typesetting industry. "
+            "Lorem Ipsum has been the industry's "
+            "standard dummy text ever since the 1500s, "
+            "when an unknown printer took a galley of "
+            "type and scrambled it to make a type specimen book.")
+
+stroka02 = set(re.split(r'\W+', stroka01))
```
Comment (vorongn) [inline src/hw04_task06_unique_words.py:17]: fixed

File: src/hw04_task02_cities.py (hunk 4 above)
Comment (vorongn) [reply src/hw04_task02_cities.py:1]: changed list names

File: src/hw04_task07_evklid.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+"""
+Даны два натуральных числа.
+Вычислите их наибольший общий делитель
+при помощи алгоритма Евклида
+(мы не знаем функции и рекурсию).
+----------------------------------
+"""
+
+number01 = int(input('число1:'))
```
Comment (belovmd) [inline src/hw04_task07_evklid.py:9]: unclear naming

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**